internal/common/tree.go

```diff
@@ -150,24 +211,26 @@ func AddChild(parent *DNSTreeNode, child *DNSTreeNode) {
 	child.Parent = parent
 }
 
-func RemoveLabelFromParents(node *DNSTreeNode, label string) {
+func removeLabelFromParents(node *DNSTreeNode, label string) {
 	if isRoot(node) {
 		return
 	}
 
-	d := findDataSetForChild(node.Parent, node.Name)
-	if d == nil {
+	eps := findEndpointsForChild(node.Parent, node.Name)
+	if len(eps) == 0 {
```

Comment:
Seems like the tests only cover `if isRoot()` part. This is strange since we don't copy stop label so it should be found only on the leafs and I would expect the situation to be opposite. Can you please explain to me what I am missing?

Reply:
ah yes, this is because the stop labels in the test are only ever directly below root. I'll add a test to cover the missed code. Thanks for pointing that out.